File: 01. Graph/01. Representataion/default.py
# Default Grapth rep contains set of (V, E) where edge is parir of vertices like (v1,v2)
# if all nodes are connected , we can represent by list of edges only

class DefaultGraph:
    def __init__(self):
        self.vertices = set()    #we can also use list/array
        self.edges = set()   #we can also use list/array

    def add_vertex(self, vertex):
        self.vertices.add(vertex)

    def add_edge(self, vertex1, vertex2):
        if vertex1 in self.vertices and vertex2 in self.vertices:
            edge = (vertex1, vertex2)
            self.edges.add(edge)
        else:
            print("One of vertex not found in the graph")


graph = DefaultGraph()
graph.add_vertex(0)
graph.add_vertex(1)
graph.add_vertex(2)
graph.add_edge(0,1)
graph.add_edge(1,2)
graph.add_edge(2,1)
print(graph.vertices)
print(graph.edges)

File: 01. Graph/01. Representataion/Default.java
import java.util.ArrayList;
import java.util.HashSet;

class DefaultGraph{
    private HashSet<Integer> vertices = new HashSet<>();
    private HashSet<ArrayList<Integer>> edges = new HashSet<>();

    public void addVertex(Integer vertex){
        vertices.add(vertex);
    }

    public void addEdge(Integer vertex1, Integer vertex2){
        if (vertices.contains(vertex1) && vertices.contains(vertex2)){
            ArrayList<Integer> edge = new ArrayList<>();
            edge.add(vertex1);
            edge.add(vertex2);
            edges.add(edge);
        }
        else{
            System.out.println("One of edge not present in the graph");
        }
    }

    public HashSet<Integer> getVertices(){
        return vertices;
    }

    public HashSet<ArrayList<Integer>> getEdges(){
        return edges;
    }
}

public class Default{
    public static void main(String args[]){
        DefaultGraph graph = new DefaultGraph();
        graph.addVertex(0);
        graph.addVertex(1);
        graph.addVertex(2);
        graph.addEdge(0,1);
        graph.addEdge(1,2);
        graph.addEdge(2, 1);
        System.out.println(graph.getVertices());
        System.out.println(graph.getEdges());

    }
}